src/main/java/com/nvidia/spark/rapids/jni/GpuTimeZoneDB.java

```diff
@@ -109,63 +122,114 @@ private static synchronized void closeResources()  {
       zoneIdToTable.clear();
       zoneIdToTable = null;
     }
-    if (fixedTransitions != null) {
-      fixedTransitions.close();
-      fixedTransitions = null;
+    if (transitions != null) {
+      transitions.close();
+      transitions = null;
     }
   }
 
-  public static ColumnVector fromTimestampToUtcTimestamp(ColumnVector input, ZoneId currentTimeZone) {
-    // TODO: Remove this check when all timezones are supported
-    // (See https://github.com/NVIDIA/spark-rapids/issues/6840)
-    if (!isSupportedTimeZone(currentTimeZone)) {
-      throw new IllegalArgumentException(String.format("Unsupported timezone: %s",
-          currentTimeZone.toString()));
+  private static long getScalefactor(ColumnVector input){
+    DType inputType = input.getType();
+    if (inputType == DType.TIMESTAMP_SECONDS){
+      return 1;
+    } else if (inputType == DType.TIMESTAMP_MILLISECONDS){
+      return 1000;
+    } else if (inputType == DType.TIMESTAMP_MICROSECONDS){
+      return 1000*1000;
     }
+    return 1;
+  }
+
+  public static boolean isSupportedTimeZone(String zoneId) {
+    try {
+      getZoneId(zoneId); // check that zoneId is a valid zone
+      return true;
+    } catch (ZoneRulesException e) {
+      return false;
+    }
+  }
+
+  // enforce that all timestamps, regardless of timezone, be less than the desired date
+  private static boolean isValidInput(ColumnVector input, ZoneId zoneId){
+    if (zoneId.getRules().isFixedOffset()){
+      return true;
+    }
+    boolean isValid = false;
+    long scaleFactor = getScalefactor(input);
+    try (Scalar targetTimestamp = Scalar.timestampFromLong(input.getType(), maxTimestamp*scaleFactor);
+         ColumnVector isGreater = input.binaryOp(BinaryOp.GREATER, targetTimestamp, DType.BOOL8)) {
+      isValid = !isGreater.any().getBoolean();
```

Comment:
This leaks the scalar value that is returned. Also if all of the input values are null, then any returns a null and the getBoolean value is invalid.